### PR TITLE
infra: use latest ruff and mypy in core

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

**Comment by Undertone0809** [comment]
Does it seem like it has changed the mypy rules?

File: libs/core/langchain_core/tracers/base.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ def on_tool_start(
             child_runs=[],
             run_type="tool",
             tags=tags or [],
-            name=name,
+            name=name,  # type: ignore[arg-type]  # TODO: fix typing
```
Comment (eyurtsev) [inline libs/core/langchain_core/tracers/base.py:339]: due to None?

File: libs/core/langchain_core/tracers/langchain_v1.py [hunk 2]
```diff
@@ -60,29 +61,30 @@ def _convert_to_v1_run(self, run: Run) -> Union[LLMRun, ChainRun, ToolRun]:
             else:
                 raise ValueError("No prompts found in LLM run inputs")
             return LLMRun(
-                uuid=str(run.id) if run.id else None,
+                uuid=str(run.id) if run.id else "",
```
Comment (eyurtsev) [inline libs/core/langchain_core/tracers/langchain_v1.py:64]: Is this safe? Is downstream code ever assuming this is a None?

File: libs/core/langchain_core/tracers/base.py (hunk 1 above)
Comment (baskaryan) [reply libs/core/langchain_core/tracers/base.py:339]: yep

File: libs/core/langchain_core/tracers/langchain_v1.py (hunk 2 above)
Comment (baskaryan) [reply libs/core/langchain_core/tracers/langchain_v1.py:64]: yea should maybe type ignore instead. cc @hinthornw any idea if None is expected downstream?

Comment (hinthornw) [reply libs/core/langchain_core/tracers/langchain_v1.py:64]: We should just delete this file entirely. It's not used anymore. @agola11 had a PR out to remove at some point that must have never been merged

**Pull request closed without merging**